### ISSUE TITLE
RuntimeError: Signal event received before subprocess transport available.

### ISSUE THREAD

**Issue by txlei** [osrf/rmf_demos#121]
running office.launch.xml on foxy and gazebo11
not sure whether the node related parameter error is a separate issue.

```
[INFO] [launch]: All log files can be found below /home/leia/.ros/log/2020-09-02-10-42-30-703418-leia-ubuntu-20-68254
[INFO] [launch]: Default logging verbosity is set to INFO
/opt/ros/foxy/lib/python3.8/site-packages/launch/frontend/expose.py:39: UserWarning: The parameter 'node_name' is deprecated, use 'name' instead
  return action_type(**kwargs)
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [xml]: Passing both 'node_name' and 'name' parameters. Only use 'name'.
Task exception was never retrieved
future: <Task finished name='Task-40' coro=<LaunchService._process_one_event() done, defined at /opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py:271> exception=RuntimeError('Signal event received before subprocess transport available.')>
Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 273, in _process_one_event
    await self.__process_event(next_event)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 293, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/opaque_function.py", line 75, in execute
    return self.__function(context, *self.__args, **self.__kwargs)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/execute_process.py", line 421, in __on_signal_process_event
    raise RuntimeError('Signal event received before subprocess transport available.')
RuntimeError: Signal event received before subprocess transport available.
[INFO] [rmf_traffic_schedule-1]: process started with pid [68256]
[INFO] [building_map_server-2]: process started with pid [68258]
[INFO] [rviz2-3]: process started with pid [68260]
[INFO] [fleet_state_visualizer-4]: process started with pid [68262]
[INFO] [building_systems_visualizer-5]: process started with pid [68264]
[INFO] [rviz2-6]: process started with pid [68266]
[INFO] [door_supervisor-7]: process started with pid [68268]
[INFO] [gzserver-8]: process started with pid [68270]
[INFO] [gzclient-9]: process started with pid [68272]
Task exception was never retrieved
future: <Task finished name='Task-41' coro=<LaunchService._process_one_event() done, defined at /opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py:271> exception=RuntimeError('Signal event received before subprocess transport available.')>
Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 273, in _process_one_event
    await self.__process_event(next_event)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 293, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/opaque_function.py", line 75, in execute
    return self.__function(context, *self.__args, **self.__kwargs)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/execute_process.py", line 421, in __on_signal_process_event
    raise RuntimeError('Signal event received before subprocess transport available.')
RuntimeError: Signal event received before subprocess transport available.
[INFO] [door_supervisor-7]: sending signal 'SIGINT' to process[door_supervisor-7]
[INFO] [rviz2-6]: sending signal 'SIGINT' to process[rviz2-6]
[INFO] [building_systems_visualizer-5]: sending signal 'SIGINT' to process[building_systems_visualizer-5]
[INFO] [fleet_state_visualizer-4]: sending signal 'SIGINT' to process[fleet_state_visualizer-4]
[ERROR] [rviz2-6]: process has died [pid 68266, exit code -2, cmd 'rviz2 -d /home/leia/rmf_demos_ws/install/demos/share/demos/include/office/office.rviz'].
[INFO] [rviz2-3]: sending signal 'SIGINT' to process[rviz2-3]
[INFO] [building_map_server-2]: sending signal 'SIGINT' to process[building_map_server-2]
[ERROR] [fleet_state_visualizer-4]: process has died [pid 68262, exit code 1, cmd '/home/leia/rmf_demos_ws/install/fleet_state_visualizer/lib/fleet_state_visualizer/fleet_state_visualizer -m L1 --ros-args --params-file /tmp/launch_params_ody2ms3b'].
[INFO] [rmf_traffic_schedule-1]: sending signal 'SIGINT' to process[rmf_traffic_schedule-1]
[ERROR] [building_systems_visualizer-5]: process has died [pid 68264, exit code 1, cmd '/home/leia/rmf_demos_ws/install/building_systems_visualizer/lib/building_systems_visualizer/building_systems_visualizer -m L1 --ros-args --params-file /tmp/launch_params_m5tl2zzc'].
[ERROR] [building_map_server-2]: process has died [pid 68258, exit code -2, cmd '/home/leia/rmf_demos_ws/install/building_map_tools/lib/building_map_tools/building_map_server /home/leia/rmf_demos_ws/install/rmf_demo_maps/share/rmf_demo_maps/office/office.building.yaml --ros-args'].
[door_supervisor-7] [INFO] [1599014550.858789824] [rclcpp]: signal_handler(signal_value=2)
[building_systems_visualizer-5] Fatal Python error: init_import_size: Failed to import the site module
[building_systems_visualizer-5] Python runtime state: initialized
[fleet_state_visualizer-4] Fatal Python error: init_import_size: Failed to import the site module
[fleet_state_visualizer-4] Python runtime state: initialized
[rmf_traffic_schedule-1] [INFO] [1599014550.867316715] [rmf_traffic_schedule_node]: Beginning traffic schedule node
[fleet_state_visualizer-4] Traceback (most recent call last):
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/site.py", line 597, in <module>
[fleet_state_visualizer-4]     main()
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/site.py", line 584, in main
[fleet_state_visualizer-4]     known_paths = addsitepackages(known_paths)
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/site.py", line 367, in addsitepackages
[fleet_state_visualizer-4]     addsitedir(sitedir, known_paths)
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/site.py", line 214, in addsitedir
[fleet_state_visualizer-4]     addpackage(sitedir, name, known_paths)
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/site.py", line 175, in addpackage
[fleet_state_visualizer-4]     exec(line)
[fleet_state_visualizer-4]   File "<string>", line 1, in <module>
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/importlib/util.py", line 14, in <module>
[fleet_state_visualizer-4]     from contextlib import contextmanager
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/contextlib.py", line 5, in <module>
[fleet_state_visualizer-4]     from collections import deque
[fleet_state_visualizer-4]   File "/usr/lib/python3.8/collections/__init__.py", line 27, in <module>
[fleet_state_visualizer-4]     from reprlib import recursive_repr as _recursive_repr
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap>", line 971, in _find_and_load_unlocked
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap>", line 914, in _find_spec
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap_external>", line 1342, in find_spec
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap_external>", line 1314, in _get_spec
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap_external>", line 1470, in find_spec
[fleet_state_visualizer-4]   File "<frozen importlib._bootstrap_external>", line 1426, in _get_spec
[fleet_state_visualizer-4] KeyboardInterrupt
[rviz2-3] [INFO] [1599014550.867699720] [rclcpp]: signal_handler(signal_value=2)
[rviz2-3] [ERROR] [1599014550.867839163] [viz]: Mirror was not initialized in enough time [120.000000s]!
[rviz2-3] Failed to publish log message to rosout: publisher's context is invalid, at /tmp/binarydeb/ros-foxy-rcl-1.1.7/src/rcl/publisher.c:395
[building_map_server-2] Traceback (most recent call last):
[building_map_server-2]   File "/home/leia/rmf_demos_ws/install/building_map_tools/lib/building_map_tools/building_map_server", line 4, in <module>
[building_map_server-2]     import re
[building_map_server-2]   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
[building_map_server-2]   File "<frozen importlib._bootstrap>", line 971, in _find_and_load_unlocked
[building_map_server-2]   File "<frozen importlib._bootstrap>", line 914, in _find_spec
[building_map_server-2]   File "<frozen importlib._bootstrap_external>", line 1342, in find_spec
[building_map_server-2]   File "<frozen importlib._bootstrap_external>", line 1314, in _get_spec
[building_map_server-2]   File "<frozen importlib._bootstrap_external>", line 1467, in find_spec
[building_map_server-2]   File "<frozen importlib._bootstrap>", line 222, in _verbose_message
[building_map_server-2] KeyboardInterrupt
[building_systems_visualizer-5] Traceback (most recent call last):
[building_systems_visualizer-5]   File "/usr/lib/python3.8/site.py", line 597, in <module>
[building_systems_visualizer-5]     main()
[building_systems_visualizer-5]   File "/usr/lib/python3.8/site.py", line 584, in main
[building_systems_visualizer-5]     known_paths = addsitepackages(known_paths)
[building_systems_visualizer-5]   File "/usr/lib/python3.8/site.py", line 367, in addsitepackages
[building_systems_visualizer-5]     addsitedir(sitedir, known_paths)
[building_systems_visualizer-5]   File "/usr/lib/python3.8/site.py", line 214, in addsitedir
[building_systems_visualizer-5]     addpackage(sitedir, name, known_paths)
[building_systems_visualizer-5]   File "/usr/lib/python3.8/site.py", line 175, in addpackage
[building_systems_visualizer-5]     exec(line)
[building_systems_visualizer-5]   File "<string>", line 1, in <module>
[building_systems_visualizer-5]   File "/usr/lib/python3.8/importlib/util.py", line 2, in <module>
[building_systems_visualizer-5]     from . import abc
[building_systems_visualizer-5]   File "/usr/lib/python3.8/importlib/abc.py", line 183, in <module>
[building_systems_visualizer-5]     class ResourceLoader(Loader):
[building_systems_visualizer-5]   File "/usr/lib/python3.8/importlib/abc.py", line 193, in ResourceLoader
[building_systems_visualizer-5]     def get_data(self, path):
[building_systems_visualizer-5]   File "/usr/lib/python3.8/abc.py", line 7, in abstractmethod
[building_systems_visualizer-5]     def abstractmethod(funcobj):
[building_systems_visualizer-5] KeyboardInterrupt
[rmf_traffic_schedule-1] [INFO] [1599014550.872657087] [rclcpp]: signal_handler(signal_value=2)
[rmf_traffic_schedule-1] [INFO] [1599014550.872884514] [rmf_traffic_schedule_node]: Closing down traffic schedule node
[rviz2-3] [WARN] [1599014550.877980496] [viz]: [rmf_traffic_ros2::~MirrorManagerFuture] Timeout while waiting for the acknowlegment of a query registration.
[gzclient-9] Gazebo multi-robot simulator, version 11.1.0
[gzclient-9] Copyright (C) 2012 Open Source Robotics Foundation.
[gzclient-9] Released under the Apache 2 License.
[gzclient-9] http://gazebosim.org
[gzclient-9] 
[gzserver-8] Gazebo multi-robot simulator, version 11.1.0
[gzserver-8] Copyright (C) 2012 Open Source Robotics Foundation.
[gzserver-8] Released under the Apache 2 License.
[gzserver-8] http://gazebosim.org
[gzserver-8] 
[rviz2-3] Failed to initialize the visualizer node
[INFO] [door_supervisor-7]: process has finished cleanly [pid 68268]
[ERROR] [rviz2-3]: process has died [pid 68260, exit code 1, cmd '/home/leia/rmf_demos_ws/install/rmf_schedule_visualizer/lib/rmf_schedule_visualizer/rviz2 -r 10 -m L1 --ros-args --params-file /tmp/launch_params_dbj3iydb'].
[INFO] [rmf_traffic_schedule-1]: process has finished cleanly [pid 68256]
[gzserver-8] [INFO] [1599014552.280639660] [teleport_dispenser]: Started TeleportDispenserPlugin node...
[gzserver-8] [INFO] [1599014552.281115024] [teleport_dispenser]: Found dispenser item: [Coke]
[gzserver-8] [INFO] [1599014552.298344478] [teleport_ingestor]: Started TeleportIngestorPlugin node...
[gzserver-8] [INFO] [1599014552.326725561] [slotcar_tinyRobot2]: Setting nominal drive speed to: 0.500000
[gzserver-8] [INFO] [1599014552.326753280] [slotcar_tinyRobot2]: Setting nominal drive acceleration to: 0.250000
[gzserver-8] [INFO] [1599014552.326760771] [slotcar_tinyRobot2]: Setting max drive acceleration to: 0.750000
[gzserver-8] [INFO] [1599014552.326767045] [slotcar_tinyRobot2]: Setting nominal turn speed to:0.600000
[gzserver-8] [INFO] [1599014552.326773478] [slotcar_tinyRobot2]: Setting nominal turn acceleration to:1.500000
[gzserver-8] [INFO] [1599014552.326779514] [slotcar_tinyRobot2]: Setting max turn acceleration to:2.000000
[gzserver-8] [INFO] [1599014552.326785777] [slotcar_tinyRobot2]: Setting stop distance to:0.750000
[gzserver-8] [INFO] [1599014552.326805302] [slotcar_tinyRobot2]: Setting stop radius to:0.750000
[gzserver-8] [INFO] [1599014552.326810761] [slotcar_tinyRobot2]: Setting tire radius to:0.100000
[gzserver-8] [INFO] [1599014552.326835281] [slotcar_tinyRobot2]: Setting base width to:0.320600
[gzserver-8] [INFO] [1599014552.326840703] [slotcar_tinyRobot2]: Setting name to: tinyRobot2
[gzserver-8] [INFO] [1599014552.333716811] [slotcar_tinyRobot2]: Initialising slotcar for tinyRobot2
[gzserver-8] [INFO] [1599014552.333827531] [slotcar_tinyRobot1]: Setting nominal drive speed to: 0.500000
[gzserver-8] [INFO] [1599014552.333838149] [slotcar_tinyRobot1]: Setting nominal drive acceleration to: 0.250000
[gzserver-8] [INFO] [1599014552.333844697] [slotcar_tinyRobot1]: Setting max drive acceleration to: 0.750000
[gzserver-8] [INFO] [1599014552.333850877] [slotcar_tinyRobot1]: Setting nominal turn speed to:0.600000
[gzserver-8] [INFO] [1599014552.333856909] [slotcar_tinyRobot1]: Setting nominal turn acceleration to:1.500000
[gzserver-8] [INFO] [1599014552.333862933] [slotcar_tinyRobot1]: Setting max turn acceleration to:2.000000
[gzserver-8] [INFO] [1599014552.333882660] [slotcar_tinyRobot1]: Setting stop distance to:0.750000
[gzserver-8] [INFO] [1599014552.333888107] [slotcar_tinyRobot1]: Setting stop radius to:0.750000
[gzserver-8] [INFO] [1599014552.333893839] [slotcar_tinyRobot1]: Setting tire radius to:0.100000
[gzserver-8] [INFO] [1599014552.333899497] [slotcar_tinyRobot1]: Setting base width to:0.320600
[gzserver-8] [INFO] [1599014552.333903299] [slotcar_tinyRobot1]: Setting name to: tinyRobot1
[gzserver-8] [WARN] [1599014552.333951812] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[gzserver-8] [INFO] [1599014552.337128062] [slotcar_tinyRobot1]: Initialising slotcar for tinyRobot1
[gzserver-8] [INFO] [1599014552.350486518] [door]: Loading DoorPlugin for [main_door]
[gzserver-8] [Msg] Waiting for master.
[gzserver-8] [Msg] Connected to gazebo master @ http://127.0.0.1:11345
[gzserver-8] [Msg] Publicized address: 192.168.1.117
[gzserver-8] Using specified value [0.5] for property [v_max_door]
[gzserver-8] Using specified value [0.3] for property [a_max_door]
[gzserver-8] Using specified value [0.15] for property [a_nom_door]
[gzserver-8] Using specified value [0.01] for property [dx_min_door]
[gzserver-8] Using specified value [500] for property [f_max_door]
[gzserver-8] Using specified attribute value [left_joint] for property [left_joint_name]
[gzserver-8] Using specified attribute value [right_joint] for property [right_joint_name]
[gzserver-8] Using specified attribute value [DoubleSwingDoor] for property [type]
[gzserver-8] Using specified attribute value [right_joint] for property [name]
[gzserver-8] Using specified value [0] for property [lower]
[gzserver-8] Using specified value [1.57] for property [upper]
[gzserver-8] Using specified attribute value [right_joint] for property [name]
[gzserver-8] Using specified value [0] for property [lower]
[gzserver-8] Using specified value [1.57] for property [upper]
[gzserver-8] Using specified attribute value [left_joint] for property [name]
[gzserver-8] Using specified value [-1.57] for property [lower]
[gzserver-8] Using specified value [0] for property [upper]
[gzserver-8] [INFO] [1599014552.355054487] [door]: Finished loading [main_door]
[gzserver-8] [WARN] [1599014552.355178371] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[gzserver-8] [INFO] [1599014552.356725309] [door]: Loading DoorPlugin for [coe_door]
[gzserver-8] Using specified value [0.5] for property [v_max_door]
[gzserver-8] Using specified value [0.3] for property [a_max_door]
[gzserver-8] Using specified value [0.15] for property [a_nom_door]
[gzserver-8] Using specified value [0.01] for property [dx_min_door]
[gzserver-8] Using specified value [500] for property [f_max_door]
[gzserver-8] Using specified attribute value [left_joint] for property [left_joint_name]
[gzserver-8] Using specified attribute value [empty_joint] for property [right_joint_name]
[gzserver-8] Using specified attribute value [SwingDoor] for property [type]
[gzserver-8] Using specified attribute value [left_joint] for property [name]
[gzserver-8] Using specified value [-1.57] for property [lower]
[gzserver-8] Using specified value [0] for property [upper]
[gzserver-8] Using specified attribute value [left_joint] for property [name]
[gzserver-8] Using specified value [-1.57] for property [lower]
[gzserver-8] Using specified value [0] for property [upper]
[gzserver-8] [INFO] [1599014552.359435644] [door]: Finished loading [coe_door]
[gzserver-8] [WARN] [1599014552.359577651] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[gzserver-8] [INFO] [1599014552.361515281] [door]: Loading DoorPlugin for [hardware_door]
[gzserver-8] Using specified value [0.5] for property [v_max_door]
[gzserver-8] Using specified value [0.3] for property [a_max_door]
[gzserver-8] Using specified value [0.15] for property [a_nom_door]
[gzserver-8] Using specified value [0.01] for property [dx_min_door]
[gzserver-8] Using specified value [500] for property [f_max_door]
[gzserver-8] Using specified attribute value [left_joint] for property [left_joint_name]
[gzserver-8] Using specified attribute value [empty_joint] for property [right_joint_name]
[gzserver-8] Using specified attribute value [SwingDoor] for property [type]
[gzserver-8] Using specified attribute value [left_joint] for property [name]
[gzserver-8] Using specified value [-1.57] for property [lower]
[gzserver-8] Using specified value [0] for property [upper]
[gzserver-8] Using specified attribute value [left_joint] for property [name]
[gzserver-8] Using specified value [-1.57] for property [lower]
[gzserver-8] Using specified value [0] for property [upper]
[gzserver-8] [INFO] [1599014552.363879555] [door]: Finished loading [hardware_door]
[ERROR] [gzclient-9]: process[gzclient-9] failed to terminate '5' seconds after receiving 'SIGINT', escalating to 'SIGTERM'
[ERROR] [gzserver-8]: process[gzserver-8] failed to terminate '5' seconds after receiving 'SIGINT', escalating to 'SIGTERM'
[INFO] [gzclient-9]: sending signal 'SIGTERM' to process[gzclient-9]
[INFO] [gzserver-8]: sending signal 'SIGTERM' to process[gzserver-8]
[ERROR] [gzserver-8]: process has died [pid 68270, exit code -15, cmd 'gzserver --verbose -s libgazebo_ros_factory.so -s libgazebo_ros_init.so /home/leia/rmf_demos_ws/install/rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world'].
[ERROR] [gzclient-9]: process has died [pid 68272, exit code -15, cmd 'gzclient --verbose /home/leia/rmf_demos_ws/install/rmf_demo_maps/share/rmf_demo_maps/maps/office/office.world'].
```

**Comment by Yadunund** [comment]
Hi,

When posting an issue here, it will be helpful to list out the branches of all the repositories being used.

Having said that I believe the issue is with not checking out the `foxy` [branch](https://github.com/osrf/rmf_core/pull/133) for `rmf_core`. We are still in the midst of officially transitioning to `foxy` and hence this PR hasn't been merged yet. Let us know if that fixes your problem.

**Comment by txlei** [comment]
able to run the office demo after solving these errors using foxy branch for a successful build
1. missing rmf_ingestor_msgs package
2. missing _dropoff_ingestor_ field in Delivery.msg for mf_task_msgs package
3. missing _table_view_ obj in rmf_traffic_ros2 (fix from #108)

one qn before closing the issue. where can i get the office loop xml for multi robot demo?

**Comment by aaronchongth** [comment]
Thanks for pointing it out! `ingestors` are features that have just been merged very recently into `master`, I will try to patch the changes into the `foxy` branches. I will start a new issue regarding this.

Do you mean [this](https://github.com/osrf/rmf_demos/blob/master/demos/launch/include/office/office_loop.launch.xml)?
